### PR TITLE
style tweaks for dl only ui

### PR DESCRIPTION
**Pull request by johngruen**
demo

fixes https://github.com/mozilla-services/screenshots/issues/3693

### CODE REVIEW

File: addon/webextension/icons/help-16.svg [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
```
Comment (johngruen) [inline addon/webextension/icons/help-16.svg]: Swapped ? out for standard photon SVG

File: addon/webextension/selector/ui.js [hunk 2]
```diff
@@ -95,16 +95,22 @@ this.ui = (function() { // eslint-disable-line no-unused-vars
     return window.downloadOnly;
   }
 
-  function renderDownloadNotice() {
-    let notice = makeEl("div", "download-only");
+  function renderDownloadNotice(initAtBottom = false) {
```
Comment (johngruen) [inline addon/webextension/selector/ui.js]: fiddled with dom structure here. added an optional param b/c the two iframes we inject this to are really different.

File: addon/webextension/selector/ui.js [hunk 3]
```diff
@@ -582,7 +588,17 @@ this.ui = (function() { // eslint-disable-line no-unused-vars
       this.bgRight.style.width = "100%";
 
       if (this.downloadNotice) {
-        this.downloadNotice.style.bottom = (docHeight - pageYOffset - winBottom + 10) + "px";
+        this.downloadNotice.style.top = (pageYOffset + winBottom - 60) + "px";
+
+        const repositionDownloadNotice = () => {
```
Comment (johngruen) [inline addon/webextension/selector/ui.js]: The download notice should be tied to the browser window, but b/c it is injected into an iframe that's the size of the document we have to do some jujitsu to make it stick

File: addon/webextension/selector/ui.js [hunk 4]
```diff
@@ -582,7 +588,17 @@ this.ui = (function() { // eslint-disable-line no-unused-vars
...
+          const currentYOffset = window.pageYOffset;
+          const currentWinBottom = window.innerHeight;
+          this.downloadNotice.style.top = (currentYOffset + currentWinBottom - 60) + "px";
+        };
+
+        const windowChangeStop = this.delayExecution(100, repositionDownloadNotice);
+        window.addEventListener('scroll', windowChangeStop);
```
Comment (johngruen) [inline addon/webextension/selector/ui.js]: Bind events to make it the download notice stick on scroll and resize events. These callbacks are only executed at a maximum every 100ms.

File: static/css/inline-selection.scss [hunk 5]
```diff
@@ -191,70 +191,84 @@ $very-light-grey: #ededed;
...
-  font-family: -apple-system, BlinkMacSystemFont, "segoe ui", "helvetica neue", helvetica, ubuntu, roboto, noto, arial, sans-serif;
-  font-weight: bold;
-  color: #fff;
-  font-size: 14px;
-  line-height: 26px;
-  white-space: nowrap;
+  transition: top 125ms ease-out, translate 125ms ease-out;
   user-select: none;
-}
+  will-change: top, translate;
```
Comment (johngruen) [inline static/css/inline-selection.scss]: not sure if this makes a difference here, but telling the browser that we expect these to move may add a bit of smoothness.

File: static/css/inline-selection.scss [hunk 6]
```diff
@@ -491,6 +505,7 @@ $very-light-grey: #ededed;
   text-align: center;
   padding-top: 20px;
   width: 400px;
+  user-select: none;
```
Comment (johngruen) [inline static/css/inline-selection.scss]: This is super important. I think it fell out of our CSS at some point. we really need to land this line.

**Comment by johngruen** [comment]
Noticed a little wiggle in that gif...that's fixed now

File: static/css/inline-selection.scss [hunk 7]
```diff
@@ -191,70 +191,87 @@ $very-light-grey: #ededed;
...
+  line-height: 22px;
+  overflow-wrap: break-word;
+  padding: 15px;
+  position: absolute;
+  right: -14px;
   white-space: normal;
-  color: #000;
-  cursor: default;
+  width: 300px;
+  z-index: 10000000000;
```
Comment (6a68) [inline static/css/inline-selection.scss]: `z-index: $overlay-z-index + 1;` is used elsewhere, seemed a little nicer?

File: addon/webextension/selector/ui.js [hunk 8]
```diff
@@ -95,16 +95,28 @@ this.ui = (function() { // eslint-disable-line no-unused-vars
     return window.downloadOnly;
   }
 
-  function renderDownloadNotice() {
-    let notice = makeEl("div", "download-only");
+  // the download notice is rendered in iframes that match the document height
+  // or the window height. If parent iframe matches widnow height, pass in true
```
Comment (6a68) [inline addon/webextension/selector/ui.js]: 'widnow'

File: addon/webextension/icons/help-16.svg (hunk 1 above)
Comment (6a68) [reply addon/webextension/icons/help-16.svg]: Derp, wish I'd realized the circled '?' was a photon thing.

File: addon/webextension/selector/ui.js [hunk 9]
```diff
@@ -95,16 +95,28 @@ this.ui = (function() { // eslint-disable-line no-unused-vars
...
+  function renderDownloadNotice(initAtBottom = false) {
+    let notice = makeEl("table", "download-only");
     notice.innerHTML = `
```
Comment (6a68) [inline addon/webextension/selector/ui.js]: idk, kinda weird to stuff divs directly inside a table element, but I guess if it works...

**Comment by 6a68** [comment]
gah, one of my main comments didn't take - look around the file for how `onScroll` is used elsewhere, you'll want to detach when the frame is hidden and check the events are trusted.

Also, the resize of the frame can expose the bottom edge of the frame, so maybe it'd be good to resize the frame itself inside the resize handler?

screen shot 2017-11-01 at 12 22 12 pm

**Comment by johngruen** [comment]
thanks _6a68 looking now

File: addon/webextension/selector/ui.js [hunk 10]
```diff
@@ -569,7 +581,7 @@ this.ui = (function() { // eslint-disable-line no-unused-vars
       this.bgTop.style.left = "0px";
       this.bgTop.style.width = "100%";
       this.bgBottom.style.top = (pos.bottom - bodyRect.top) + "px";
-      this.bgBottom.style.height = docHeight - (pos.bottom - bodyRect.top) + "px";
+      this.bgBottom.style.height = "100vh";
```
Comment (johngruen) [inline addon/webextension/selector/ui.js]: @6a68 turns out this fixes the overlay issue you saw on example.com

**Comment by johngruen** [comment]
@6a68 i think i hit all of your concerns, event handlers are now removed with the box iframe, and i even fixed that (preexisting) resize issue!

**Comment by 6a68** [comment]
sorry for the delay in the followup review. merging 🥂

**Pull request merged**